src/BasePlatform.ts

```diff
@@ -258,7 +258,9 @@ export default abstract class BasePlatform {
         return null;
     }
 
-    setLanguage(preferredLangs: string[]) {}
+    async setLanguage(preferredLangs: string[]) {
+        throw new Error("Unimplemented");
+    }
```

Comment:
should this not just be an abstract to know that its not implemented at compile time rather than at runtime?